numpy/lib/tests/test_polynomial.py

```diff
@@ -1,93 +1,87 @@
->>> p.integ(5)
-poly1d([ 0.00039683,  0.00277778,  0.025     ,  0.        ,  0.        ,
-        0.        ,  0.        ,  0.        ])
->>> p.deriv()
-poly1d([ 2.,  2.])
->>> p.deriv(2)
-poly1d([ 2.])
-
->>> q = np.poly1d([1.,2,3], variable='y')
->>> print(q)
-   2
-1 y + 2 y + 3
->>> q = np.poly1d([1.,2,3], variable='lambda')
->>> print(q)
-        2
-1 lambda + 2 lambda + 3
-
->>> np.polydiv(np.poly1d([1,0,-1]), np.poly1d([1,1]))
-(poly1d([ 1., -1.]), poly1d([ 0.]))
-
-'''
-import numpy as np
-from numpy.testing import (
-    run_module_suite, assert_, assert_equal, assert_array_equal,
-    assert_almost_equal, assert_array_almost_equal, assert_raises, rundocs
-    )
-
+from __future__ import absolute_import, division, print_function
 
-class TestDocs(object):
-    def test_doctests(self):
-        return rundocs()
+import numpy as np
+from numpy.testing import (assert_, assert_almost_equal,
+                           assert_array_almost_equal, assert_array_equal,
+                           assert_equal, assert_raises, run_module_suite)
+
+
+def assert_str_equal(actual, desired, err_msg='', verbose=True):
+    assert_equal(str(actual), desired, err_msg, verbose)
+
+
+def assert_repr_equal(actual, desired, err_msg='', verbose=True):
+    assert_equal(repr(actual), desired, err_msg, verbose)
+
+
+class TestPolynomial(object):
+    def test_poly1d_str_and_repr(self):
+        # GH9416: Convert doctests to regular tests.
+        p = np.poly1d([1., 2, 3])
+        assert_repr_equal(p, 'poly1d([1., 2., 3.])')
+        assert_str_equal(p,
+                         '   2\n'
+                         '1 x + 2 x + 3')
+
+        q = np.poly1d([3., 2, 1])
+        assert_repr_equal(q, 'poly1d([3., 2., 1.])')
+        assert_str_equal(q,
+                         '   2\n'
+                         '3 x + 2 x + 1')
+
+        r = np.poly1d([1.89999 + 2j, -3j, -5.12345678, 2 + 1j])
+        assert_str_equal(r,
+                         '            3      2\n'
+                         '(1.9 + 2j) x - 3j x - 5.123 x + (2 + 1j)')
+
+        assert_str_equal(np.poly1d([-3, -2, -1]),
+                         '    2\n'
+                         '-3 x - 2 x - 1')
+
+    def test_poly1d_resolution(self):
+        p = np.poly1d([1., 2, 3])
+        q = np.poly1d([3., 2, 1])
+        assert_equal(p(0), 3.0)
+        assert_equal(p(5), 38.0)
+        assert_equal(q(0), 1.0)
+        assert_equal(q(5), 86.0)
+
+    def test_poly1d_math(self):
+        # here we use some simple coeffs to make calculations easier
+        p = np.poly1d([1., 2, 4])
+        q = np.poly1d([4., 2, 1])
+        assert_equal(p/q, (np.poly1d([0.25]), np.poly1d([1.5, 3.75])))
+        assert_equal(p.integ(), np.poly1d([1/3, 1., 4., 0.]))
+        assert_equal(p.integ(1), np.poly1d([1/3, 1., 4., 0.]))
+
+        p = np.poly1d([1., 2, 3])
+        q = np.poly1d([3., 2, 1])
+        assert_array_equal(p * q, np.poly1d([3., 8., 14., 8., 3.]))
```

Comment:
Why is this one `assert_array_equal`? I think you want `assert_equal` for all of them